scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -203,14 +203,13 @@ sealed trait Execution[+T] extends java.io.Serializable { self: Product =>
   override val hashCode: Int = ScalaRunTime._hashCode(self)
 
   override def equals(other: Any): Boolean = {
-    //removed from Scala 2.12
-    def _equals(x: Product, y: Any): Boolean = y match {
-      case y: Product if x.productArity == y.productArity =>
-        x.productIterator sameElements y.productIterator
+    val productEquals = other match {
+      case p: Product if self.productArity == p.productArity =>
```

Comment:
nit: I guess `hashCode` check was made before `_equals` call for perf reasons. Therefore this code has slightly worse performance if it matters.

I recommend  just remove separate `hashCode == other.hashCode` check and put it into `if` part.